src/clj/rems/db/applications.clj

```diff
@@ -52,9 +52,11 @@
   (let [id-prefix (str (.getYear time))]
     (format-external-id (allocate-external-id! id-prefix))))
 
-(defn application-created-event [{:keys [application-id catalogue-item-ids time actor allocate-external-id?]}]
+(defn application-created-event [{:keys [application-id catalogue-item-ids time actor allocate-external-id?]
```

Comment:
How about keeping this function pure and always passing in the application-id? That way it would suffice to inject `db/create-application!` and always use this pure function to generate the created event.

Reply:
Hmm okay this function is fetching a lot of stuff from the db. I wonder if it could be turned into a function in `rems.application.commands` that takes the injections map as a parameter.

Reply:
It's a complex beast.